boot/kernel_os.go

```diff
@@ -63,7 +64,12 @@ func (*coreKernel) IsTrivial() bool { return false }
 
 func (k *coreKernel) RemoveKernelAssets() error {
 	// XXX: shouldn't we check the snap type?
-	bootloader, err := bootloader.Find("", nil)
+	opts := &bootloader.Options{}
+	if k.dev.HasModeenv() {
+		// we are in uc20 mode, so find an extractable kernel
+		opts.ExtractedRunKernelImage = true
```

Comment:
afaict this bit is not tested here, nor in the larger PR

Reply:
hmm I suppose this method isn't called in the CoreKernel test here: https://github.com/snapcore/snapd/pull/8001/files#diff-bf7a36650e7c3d6180b9547be1e9712eR374-R396

I will add something tomorrow if @mvo5 doesn't beat me to it :-)

Reply:
I've cherry-picked, edited and pushed fafd3b526fc15061a4900c62b5e9abffcfd0078b.